src/WebJobs.Script.WebHost/DependencyInjection/DependencyValidator/DependencyValidator.cs

```diff
@@ -53,11 +53,11 @@ private static ExpectedDependencyBuilder CreateExpectedDependencies()
                 .Optional<FuncAppFileProvisioningService>() // Used by powershell.
                 .Optional<JobHostService>() // Missing when host is offline.
                 .Optional<FunctionsSyncService>() // Conditionally registered.
+                .Optional<AzureMonitorDiagnosticLoggerProvider>() // Conditionally registered.
```

Comment:
You can just remove this completely. The `ExpectSubcollection` below means -- we require these services, but customers (or us) can add additional services without failing the check. So just by removing the `.Expect<AzureMonitorDiagnosticLoggerProvider>()`, that puts this into an "optional" category.